envoy/http/async_client.h

```diff
@@ -358,6 +371,16 @@ class AsyncClient {
     bool is_shadow{false};
 
     bool is_shadow_suffixed_disabled{false};
+
+    // The parent span that child spans are created under to trace egress requests/responses.
+    // If not set, requests will not be traced.
+    Tracing::Span* parent_span_{nullptr};
+    // The name to give to the child span that represents the async http request.
+    // If left empty and parent_span_ is set, then the default name will have the cluster name.
+    // Only used if parent_span_ is set.
+    std::string child_span_name_{""};
+    // Sampling decision for the tracing span. The span is sampled by default.
+    absl::optional<bool> sampled_{true};
```

Comment:
Now, because these fields were moved to StreamOptions from RequestOptions, we should handle these fields in the `operator==` of StreamOptions.